serving/samples/jib-app-java/README.md

```diff
@@ -0,0 +1,99 @@
+# Jib Sample App
+
+A sample app that demonstrates using [Jib](https://github.com/GoogleContainerTools/jib) to build Java applications
+and running with Knative Serving.
+
+Jib can package any Java application without using the Dockerfile.
+
+This deploys the [Helloworld Java](https://github.com/knative/docs/tree/master/serving/samples/helloworld-java)
+sample app for Knative, and builds without using its Dockerfile.
+
+
+## Prerequisites
+
+* [Install Knative Serving](../../../install/README.md)
+* [Configure Image Push Credentials](../../../build/auth.md#basic-authentication-docker)
+
+## Running
+
+This sample uses the [Jib build
+template](https://github.com/knative/build-templates/tree/master/jib)
+in the [build-templates](https://github.com/knative/build-templates/) repo.
+
+Install the Jib Maven build template:
+
+```shell
+kubectl apply -f https://raw.githubusercontent.com/knative/build-templates/master/jib/jib-maven.yaml
+```
+
+Then you can deploy this to Knative Serving from the root directory
+by entering the following commands:
+
+```shell
+export REPO="your/registry"
+
+# For example, with Google Container Registry:
+# export REPO="gcr.io/YOUR_PROJECT_ID"
+
+perl -pi -e "s@DOCKER_REPO_OVERRIDE@$REPO@g" sample.yaml
```

Comment:
I'm not sure which tools we assume here; you might need to swich the `perl -pie` with some other text. @samodell can probably help.

Reply:
this is also what the other repository has done as well. at least it's consistent.